pyatlan/model/search.py

```diff
@@ -1893,8 +1905,21 @@ def validate_query(cls, v, values):
         else:
             raise ValueError("Either query or post_filter is required")
 
+    # Transforming dictionaries into SortItem instances to facilitate deserialization.
+    # This conversion occurs prior to invoking the initializer and validator_sort(),
+    # preventing potential validation-related issues from arising.
```

Comment:
I would simplify it and include it as a method docstring:

```suggestion
"""
Convert dictionary entries into SortItem instances before validation  
to ensure proper deserialization and prevent validation issues.
"""
```

Reply:
Done